### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Hipst0man**

### CODE REVIEW

File: AnyType.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+AnyType::AnyType(const AnyType& copy)
+{
+    data = copy.data;
+    type_hash = copy.type_hash;
+}
+
+AnyType::AnyType(AnyType&& copy)
+{
+    data = copy.data;
```
Comment (Stepbura) [inline AnyType.cpp]: Have you considered to use std::move for move constructor

File: AnyType.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+void AnyType::swap(AnyType& right)
+{
+    const AnyType tmp(*this);
+    *this = right;
+    right = tmp;
+}
+
+std::string AnyType::dataType()
+{
+    if (type_hash == 0) return "There is no stored value";
```
Comment (Stepbura) [inline AnyType.cpp:150]: it is better use swith-case here

**Review by Stepbura — CHANGES_REQUESTED** [review]
Please take a look my comments

File: AnyType.cpp (hunk 2 above)
Comment (Hipst0man) [reply AnyType.cpp:150]: Hash_code for the same type can change between invocations of the same program. 
Swith-case requires a constant value. That is why i use if-else-if.

File: AnyType.cpp (hunk 1 above)
Comment (Hipst0man) [reply AnyType.cpp]: Yes, but my class consists of built-in types, so std::move is an equivalent to copying.

**Pull request closed without merging**